### DISCUSSION TITLE
useEffect is triggered everytime when having form error as a dependency

### DISCUSSION THREAD

**Discussion by AsuraKev** [react-hook-form/react-hook-form#11599] [Q&A]
Hi guys,

Does anyone know why below is triggering the useEffect to run even when the error object are exactly the same.? When i click submit twice, I didnt expect the effect to run because the errors are the same..

thanks! :)

  const {
    formState: { errors: validationErrors, isDirty, isSubmitting, isValid },
    getValues,
    setValue,
    handleSubmit,
    reset,
    watch,
    control,
  } = useForm<NewConfigurationFormData>({
    mode: 'all',
    resolver: zodResolver(validationSchema),
    defaultValues: defaultValue,
  });

  useEffect(() => {
    alert();
  }, [validationErrors]);

**Comment by sujinleeme** [comment]
Could you provide more detailed code including `validationSchema` and react components?

**Reply by AsuraKev** [reply]
I have created a codepen to replicate -> https://codesandbox.io/p/sandbox/react-hook-form-get-started-ts-forked-ryyt8f?file=%2Fsrc%2Findex.tsx%3A11%2C33

Try the following actions:

1. Try clear the input, it ll trigger validation. The effect didnt run tho event there is validation error
2. Try clear the input and keep clicking the submit button. The effect fires everytme the submit is clicked even though error object is exactly the same

thanks! :)

**Reply by sujinleeme** [reply]
The codesandbox link is not found. Can you reshare?

**Reply by AsuraKev** [reply]
sorry, just made it public -> https://codesandbox.io/p/sandbox/react-hook-form-get-started-ts-forked-ryyt8f?file=%2Fsrc%2Findex.tsx
